app/lib/get-workflows-in-order.js

```diff
@@ -0,0 +1,52 @@
+function getWorkflowsInOrder(project, query) {
+  query = query || {};
+  const order = (project.configuration && project.configuration.workflow_order) ?
+    project.configuration.workflow_order :
+    [];
+
+  // TODO remove default page_size once pagination solution implemented
+  query.page_size = query.page_size || 20;
+
+  function getAllWorkflows(query) {
+    let allWorkflows = [];
+    return getWorkflows(query, 1);
+
+    function getWorkflows(query, page) {
+      return project.get('workflows', Object.assign({}, query, { page }))
+        .then((workflows) => {
+          allWorkflows = allWorkflows.concat(workflows);
+          const meta = workflows[0] ? workflows[0].getMeta() : null;
+          if (meta && meta.page !== meta.page_count) {
+            return getWorkflows(query, page + 1);
+          }
+          return Promise.resolve(allWorkflows);
+        })
+        .catch((error) => {
+          console.info(error);
+          return Promise.resolve(allWorkflows);
```

Comment:
could this allWorkflows promise return and the other above be consolidated to a final `.then` chain block after the catch? I.e. always return the final built ordered workflows array.

Reply:
Sorry, but I'm not following you here. This catches any errors in the `then` block, which won't happen if we move the `then` block after the catch. It's meant to ensure that the promise chain always returns something, even if one of the recursively chained promises fails.

Reply:
Oh wait, do you mean

```
const allWorkflows;
getWorkflows()
.then( workflows => // maybe get some more workflows and update allWorkflows)
.catch( error => // well, that didn't work )
.then( () => // return the final state of allWorkflows)
```

I can try that and see if it works.